lib/index.js

```diff
@@ -26,18 +43,62 @@ module.exports = function (options, callback) {
       }
     })
 
+    async.parallel([
+      loadPlugins.bind(null, server, config, couchConfig),
+      bundleHoodieClient.bind(null, config)
+    ], function (error) {
+      callback(error, server, config)
+    })
+  })
+}
+
+function loadPlugins (server, config, couchConfig, callback) {
+  require('./hapi')(config, couchConfig, function (error, plugins) {
+    /* istanbul ignore next */
+    if (error) {
+      return callback(error)
+    }
+
     log.silly('hapi', 'Registering internal plugins')
-    require('./hapi')(config, couchConfig.authentication_db, function (err, plugins) {
-      /* istanbul ignore next */
-      if (err) return callback(err)
 
-      server.register(plugins, function (err) {
-        /* istanbul ignore next */
-        if (err) return callback(err)
+    server.register(plugins, function (error) {
+      if (error) {
+        return callback(error)
+      }
+
+      server.plugins.account.api.accounts.on('add', userDatabases.add.bind(null, config, server))
+      server.plugins.account.api.accounts.on('remove', userDatabases.remove.bind(null, config, server))
+
+      log.info('hapi', 'plugins registered')
+      callback(null, server, config)
+    })
+  })
+}
+
+function bundleHoodieClient (config, callback) {
+  var hoodieClientModulePath = path.dirname(require.resolve('hoodie-client/package.json'))
+  var hoodieClientPath = path.join(hoodieClientModulePath, 'dist/hoodie.js')
+  var bundleTargetPath = path.resolve('./.hoodie/client.js')
+
+  // https://github.com/hoodiehq/hoodie-client/issues/34
+  // var hoodieMinPath = path.join(hoodieClientModulePath, 'dist/hoodie.min.js')
+
+  log.silly('bundle', 'bundling ' + hoodieClientPath + ' into ' + bundleTargetPath)
+
+  var stream = fs.createReadStream(hoodieClientPath)
+  stream.pipe(fs.createWriteStream(bundleTargetPath))
+  stream.on('error', callback)
+  stream.on('end', function () {
+    // TODO: pass client configuration to constructor
+    fs.appendFile(bundleTargetPath, '\n\nhoodie = new Hoodie()', function (error) {
```

Comment:
here is where we can pass in configuration to the client in future, like `new Hoodie({url: "https://my.hoodieapp.com"})`